packages/frontend/src/hooks/tableVisualization/useTableConfig.ts

```diff
@@ -158,13 +162,26 @@ const useTableConfig = (
         [columnProperties],
     );
 
-    const canUsePivotTable =
+    const isPivotTableEnabled =
         resultsData?.metricQuery &&
         resultsData.metricQuery.metrics.length > 0 &&
         resultsData.rows.length &&
         pivotDimensions &&
         pivotDimensions.length > 0;
 
+    const dimensions = useMemo(() => {
+        if (!itemsMap) return [];
+
+        return columnOrder.filter((fieldId) => {
+            const item = itemsMap[fieldId];
+            return item && isField(item)
+                ? item.fieldType === FieldType.DIMENSION
+                : false;
+        });
+    }, [columnOrder, itemsMap]);
+
+    const canUseSubtotals = dimensions.length > 1;
```

Comment:
can we use `resultsData && resultsData.metricQuery.dimensions.length > 0` instead? if so we could avoid `useMemo` above

Reply:
`resultsData.metricQuery.dimensions` does not update when dimensions are removed, so if all but one dimension is removed and "Configure" is chosen then the "Show subtotals" checkbox will still be enabled because `resultsData.metricQuery.dimensions` still has the original number of dimensions while the `dimensions` constant calculated here reflects the change to one dimension.

Reply:
that makes sense! `useTableConfig` hook is so confusing!